packages/xstate-wallet/src/workflows/create-and-fund-ledger.ts

```diff
@@ -0,0 +1,173 @@
+  initial: 'isChannelIdDefined',
+  on: {[ActionTypes.ErrorCustom]: {target: 'failure'}},
+  states: {
+    isChannelIdDefined: {
+      on: {
+        '': [{cond: 'doesChannelIdExist', target: 'supportPreFundState'}, {target: 'initialize'}]
+      }
+    },
+    initialize: {
+      initial: 'initializeChannel',
+
+      states: {
+        initializeChannel: {
+          invoke: {
+            src: 'initializeChannel',
+            onDone: {target: 'createObjective', actions: ['assignChannelId']}
+          }
+        },
+        createObjective: {
+          invoke: {
+            src: 'createObjective',
+
+            onDone: [{target: 'done'}]
+          }
+        },
+        done: {type: 'final'}
+      },
+      onDone: 'supportPreFundState'
+    },
+
+    supportPreFundState: getDataAndInvoke<WorkflowContext, any>(
+      {src: 'getPreFundState'},
+      {src: 'supportState'},
+      'fundChannel'
+    ),
+
+    fundChannel: getDataAndInvoke<WorkflowContext, any>(
+      {src: 'getDepositingInfo'},
+      {src: 'depositing'},
+      'done'
+    ),
+
+    done: {type: 'final'},
+    failure: {type: 'final'}
+  }
+};
+
+const initializeChannel = (
+  store: Store
+): WorkflowServices['initializeChannel'] => async context => {
+  const stateVars: StateVariables = {
+    outcome: context.initialOutcome,
+    turnNum: bigNumberify(0),
+    isFinal: false,
+    appData: '0x0'
+  };
+  const entry = await store.createChannel(context.participants, CHALLENGE_DURATION, stateVars);
+  await store.setFunding(entry.channelId, {type: 'Direct'});
+  await store.setLedger(entry.channelId);
+  return entry.channelId;
+};
+
+const createObjective = (store: Store): WorkflowServices['createObjective'] => async context => {
+  const objective: Objective = {
+    type: 'FundLedger',
+    participants: context.participants,
+    data: {ledgerId: context.ledgerId}
+  };
+  return store.addObjective(objective);
+};
+const getPreFundState = (store: Store): WorkflowServices['getPreFundState'] => async context => {
+  const {latestState} = await store.getEntry(context.ledgerId);
+  return {state: latestState};
+};
+// TODO: Can this be shared with create-and-fund?
+const getDepositingInfo = (
+  store: Store
+): WorkflowServices['getDepositingInfo'] => async context => {
+  const {supported, myIndex} = await store.getEntry(context.ledgerId);
+  const {allocationItems} = checkThat(supported?.outcome, isSimpleEthAllocation);
+
+  const fundedAt = allocationItems.map(a => a.amount).reduce(add);
+  let depositAt = bigNumberify(0);
+  for (let i = 0; i < allocationItems.length; i++) {
+    const {amount} = allocationItems[i];
+    if (i !== myIndex) depositAt = depositAt.add(amount);
+    else {
+      const totalAfterDeposit = depositAt.add(amount);
+      return {channelId: context.ledgerId, depositAt, totalAfterDeposit, fundedAt};
+    }
+  }
+
+  throw Error(`Could not find an allocation for participant id ${myIndex}`);
```

Comment:
This could be shoehorned to work, something like:

```
    return await CreateAndFund.getDepositingInfo({channelId: context.ledgerId, allocations: allocationItems});
```

though, you end up making two calls to the database.

Reply:
I think I prefer to keep it separate for now so any changes in `CreateAndFund.getDepositingInfo` don't break an unrelated workflow.